arch/arm/core/aarch32/cortex_m/vector_table.S

```diff
@@ -72,11 +72,11 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
     .word 0
     .word z_arm_pendsv
 #if defined(CONFIG_CPU_CORTEX_M_HAS_SYSTICK)
-#if defined(CONFIG_SYS_CLOCK_EXISTS) && defined(CONFIG_CORTEX_M_SYSTICK)
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
```

Comment:
Can you add in inline comment to prevent accidental changes?
E.g. 
```
/* Install z_clock_isr even if CORTEX_M_SYSTICK is not
 * set (e.g. to support out-of-tree SysTick-based timer drivers)
 */
```

or
`/* Do not check for CORTEX_M_SYSTICK, to support out-of-tree SysTick-based timer drivers) */`